### DISCUSSION TITLE
Programmatic access to bpaf's usage message?

### DISCUSSION THREAD

**Discussion by crispinb** [pacak/bpaf#186] [Q&A] [answered]
I'd like to grab bpaf's own usage message (ie. not a custom one) for my own use. 

This is mainly because I want a `help` subcommand which (a)  does the same as the standard `--help`, and (b) will be the main fallback (ie. just invoking my command with zero args will run this `help` command, thus displaying the standard usage message).

Is this possible? Or otherwise is there a better way of doing the above?

**Comment by crispinb** [comment]
Sorry this should have been a discussion.

**Comment by pacak** [comment, marked as answer]
In bpaf terms subcommand is something with a dedicated name and help so the only thing `help` subcommand can display is a help for itself which is not very useful, you can return a default enum variant [like so](https://github.com/pacak/bpaf/blob/master/examples/derive_this_or_that.rs).

Next, to display the help message in your own code you can either use [`get_usage`](https://docs.rs/bpaf/0.7.10/bpaf/batteries/fn.get_usage.html) or implement your own variant of it - batteries only use public API so you can do that too.

In general capturing all the arguments from OS, adding extra `--help` at the end and passing it to `Args::run_inner` will give you help message

**Reply by crispinb** [reply]
Thanks, I found `get_usage`, which (a) does what I need, and (b) made me realise it would have been very straightforward to implement myself!
